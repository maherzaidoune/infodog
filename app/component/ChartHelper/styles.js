import { StyleSheet } from 'react-native';
import { colors } from '../../constants/colors';

export const styles = StyleSheet.create({
  container: {
    height: 50,
    alignItems: 'flex-end',
    justifyContent: 'center',
    paddingEnd: 20,
    backgroundColor: colors.secondary,
    marginBottom: 5,
  },
});
