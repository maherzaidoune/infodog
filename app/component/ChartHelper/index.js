import React, {useState} from 'react';
import {View, Text, Platform, TouchableOpacity} from 'react-native';
import Dialog from 'react-native-dialog';
import {colors} from '../../constants/colors';
import {Strings} from '../../constants/Strings';
import {styles} from './styles';

const ChartHelper = ({frequency = 1, update = () => {}}) => {
  const [dialogVisible, setDialogVisible] = useState(false);
  const [value, setValue] = useState(frequency);
  const updateFrequency = newFrequency => {
    setDialogVisible(false);
    if (!newFrequency || isNaN(newFrequency)) {
      return;
    }
    update(newFrequency * 1000);
  };
  return (
    <>
      <View style={styles.container}>
        <TouchableOpacity
          activeOpacity={0.1}
          onPress={() => setDialogVisible(true)}>
          <Text>{`every ${frequency} seconds`}</Text>
        </TouchableOpacity>
      </View>
      <Dialog.Container visible={dialogVisible}>
        <Dialog.Title>{Strings.UPDATE_FREQUENCY_TITLE}</Dialog.Title>
        <Dialog.Input
          keyboardType={'number-pad'}
          placeholder={'Frequency in seconds'}
          autoFocus={Platform.OS === 'ios'}
          onChangeText={text => setValue(text)}
        />
        <Dialog.Button
          label={Strings.CANCEL}
          onPress={() => setDialogVisible(false)}
        />
        <Dialog.Button
          label={Strings.UPDATE}
          onPress={() => updateFrequency(value)}
        />
      </Dialog.Container>
    </>
  );
};

export default React.memo(ChartHelper);
