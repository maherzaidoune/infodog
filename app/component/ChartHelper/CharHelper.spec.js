import React from 'react';

import renderer from 'react-test-renderer';
import ChartHelper from '.';

test('ChartHelper renders correctly', () => {
  const tree = renderer.create(<ChartHelper frequency={1}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
