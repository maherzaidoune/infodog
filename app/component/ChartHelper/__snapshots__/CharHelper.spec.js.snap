// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartHelper renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#e6e6e6",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 10,
        "paddingEnd": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingEnd": 5,
          }
        }
      >
        every 1 seconds
         
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#774aa4",
                "fontSize": 17,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï€“
        </Text>
         
      </Text>
    </View>
  </View>,
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onHide={[Function]}
    renderToHardwareTextureAndroid={true}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </Modal>,
]
`;
